metrix/src/app.rs

```diff
@@ -99,22 +117,39 @@ fn aggregate_metrics_route(
 
     let bucket_size = (end_timestamp - start_timestamp) as f32 / bucket_count as f32;
 
+    let aggregate: String;
+    if aggregation.is_some() {
+        let result = build_query_aggregate(aggregation.unwrap(), &parameter_name);
+        match result {
+            Ok(o) => {
+                aggregate = o;
+            },
+            Err(_) => {
+                return Err(BadRequest(Some("Unknown aggregate type".to_string())))
+            }
+        }
+    } else {
+        return Err(BadRequest(Some("No aggregate type provided".to_string())))
+    }
+
     let query_string = format!(
         "SELECT
-            COUNT(*) as value,
+            ({}::BIGINT) as value,
```

Comment:
what if it's some puny decimal ? 🤔

Reply:
Completely valid point. That actually bring up a good point of what data type we should be trying to squeeze these results into. f64 might just be the safest one to use, even if in some cases it'll be less mem efficient. At least it will suit all cases.

I'll try making this either DECIMAL or NUMERIC, and squeezing it into `f64`

Reply:
Went with DOUBLE since it can be squished into a rust primitive (f64). We can get arbitrary precision and size with NUMERIC or DECIMAL and  BigDecimal, but we'll cross that bridge when we come to it.